### ISSUE TITLE
DecimalType loss of precision

### ISSUE THREAD

**Issue by barticus** [ChilliCream/hotchocolate#1929]
**Describe the bug**
Decimal scalar values lose precision due to them being serialized into scientific format from the request in 10.4.3

**To Reproduce**
1. Have an input field with a Decimal type in your schema
2. Make a request with a (not too large) number with a decimal place, e.g. 200000.99
3. The value resolved will be rounded up, e.g. 200001.00

**Expected behavior**
The full value is passed through

**Additional context**
I've fixed this locally by changing the "SerializeDecimal" method in the DecimalType class to:
```
private static string SerializeDecimal(decimal value) =>
            value.ToString("G", CultureInfo.InvariantCulture);
```

Similar to https://github.com/ChilliCream/hotchocolate/issues/1846

**Comment by PascalSenn** [comment]
@barticus  this make sense.

Default precision specifier of "E" is : 6.
https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings?view=netcore-3.1

Where as "G" is the number of significant digits.

Do you like to provide a PR to version_10_0_0_master and master?

**Comment by barticus** [comment]
Thanks @PascalSenn .
I didn't get a good chance to dig out those different format specifiers so thanks for the link. Having a look, is "G" your preferred option, or "F"?

**Comment by michaelstaib** [comment]
I would say 'F'

**Comment by michaelstaib** [comment]
We decided for `G` in the end and this fix is now going into version 10.5.0-preview.8
